packages/main/cypress/specs/DateRangePicker.cy.tsx

```diff
@@ -0,0 +1,570 @@
+import "../../dist/Assets.js";
+import { setLanguage } from "@ui5/webcomponents-base/dist/config/Language.js";
+import DateRangePicker from "../../src/DateRangePicker.js";
+
+describe("DateRangePicker general interaction", () => {
+	afterEach(() => {
+		// eslint-disable-next-line
+		cy.wait(200);
+	});
+	it("Custom Validation Error", () => {
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("invalid input")
+			.realPress("Enter");
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value-state", "Negative");
+	});
+
+	it("Custom Validation None", () => {
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("09/09/2020 - 10/10/2020")
+			.realPress("Enter");
+
+		cy.get("@dateRangePicker")
+			.should("have.attr", "value-state", "None");
+	});
+
+	it("Selected dates are updated after value update in the input field", () => {
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		const timestamp_9_Sep_2020 = 1599609600;
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("09/09/2020 - 10/10/2020")
+			.realPress("Enter");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.shadow()
+			.find("ui5-calendar")
+			.shadow()
+			.find("ui5-daypicker")
+			.invoke("attr", "timestamp")
+			.should("eq", timestamp_9_Sep_2020.toString());
+	});
+
+	it("Is delimiter set", () => {
+		cy.mount(<DateRangePicker formatPattern="MMM d, y" delimiter="@"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
+			.realType("Feb 25, 2022 @ Feb 28, 2022")
+			.realPress("Enter");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.invoke("attr", "delimiter", "###")
+			.should("have.attr", "delimiter", "###");
+
+		cy.get<DateRangePicker>("@dateRangePicker")
+			.should("have.attr", "value", "Feb 25, 2022 ### Feb 28, 2022");
+	});
+
+	it("startDateValue and endDateValue getter", () => {
+		cy.mount(<DateRangePicker formatPattern="dd/MM/yyyy"></DateRangePicker>);
+
+		cy.get<DateRangePicker>("[ui5-daterange-picker]")
+			.as("dateRangePicker")
+			.shadow()
+			.find("ui5-input")
+			.realClick()
```

Comment:
i think it will be good to ensure that the input is focused after the click and before we type